site/app/templates/submission/homework/SubmitBox.twig

```diff
@@ -15,6 +15,11 @@
         {% if has_due_date %}
             <h2>Due: {{ formatted_due_date }}</h2>
         {% endif %}
+        <div id="loading_bar">
```

Comment:
Instead of creating our own progress bar lets use the HTML native `<progress id="exampleid" value="0" ></progress>`

That way you can simply upadte the values like
```
progressBar.max = e.total;
progressBar.value = e.loaded;
```
where `progressBar` is the progress element.

That way you won't have to caclulate and update the html and css in the JS.

Reply:
Hello @shailpatels, thanks, I updated it in the new commits below.